components/ai_chat/browser/ai_chat_tab_helper.cc

```diff
@@ -0,0 +1,441 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/ai_chat/browser/ai_chat_tab_helper.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "base/functional/bind.h"
+#include "base/memory/weak_ptr.h"
+#include "base/ranges/algorithm.h"
+#include "base/strings/strcat.h"
+#include "base/strings/string_split.h"
+#include "base/strings/string_util.h"
+#include "brave/components/ai_chat/browser/constants.h"
+#include "brave/components/ai_chat/browser/page_content_fetcher.h"
+#include "brave/components/ai_chat/common/ai_chat.mojom-shared.h"
+#include "brave/components/ai_chat/common/pref_names.h"
+#include "components/grit/brave_components_strings.h"
+#include "components/prefs/pref_service.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/navigation_entry.h"
+#include "content/public/browser/storage_partition.h"
+#include "ui/base/l10n/l10n_util.h"
+
+using ai_chat::mojom::CharacterType;
+using ai_chat::mojom::ConversationTurn;
+using ai_chat::mojom::ConversationTurnVisibility;
+
+namespace ai_chat {
+
+AIChatTabHelper::AIChatTabHelper(content::WebContents* web_contents)
+    : content::WebContentsObserver(web_contents),
+      content::WebContentsUserData<AIChatTabHelper>(*web_contents),
+      pref_service_(
+          *user_prefs::UserPrefs::Get(web_contents->GetBrowserContext())) {
+  ai_chat_api_ =
+      std::make_unique<AIChatAPI>(web_contents->GetBrowserContext()
+                                      ->GetDefaultStoragePartition()
+                                      ->GetURLLoaderFactoryForBrowserProcess());
+  // TODO(petemill): Observe opt-in - don't send any requests before opt-in
```

Comment:
Maybe better to resolve this before it's in the codebase and we find some surprise down the line

Reply:
resolved